include/eve/module/core/regular/impl/simd/arm/sve/has_equal_in.hpp

```diff
@@ -0,0 +1,98 @@
+  //==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/conditional.hpp>
+#include <eve/concept/value.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/detail/implementation.hpp>
+#include <eve/module/core/regular/is_equal.hpp>
+#include <eve/module/core/regular/combine.hpp>
+#include <eve/module/core/regular/rotate.hpp>
+
+namespace eve::detail
+{
+  template<callable_options O, typename T, typename N, typename Pred>
+  constexpr EVE_FORCEINLINE logical<wide<T, N>> has_equal_in_(EVE_REQUIRES(sve2_), O const& opts, wide<T, N> x, wide<T, N> match_against, Pred op) noexcept
+    requires sve_abi<abi_t<T, N>>
+  {
+    constexpr auto c = categorize<wide<T, N>>();
+    if constexpr (std::same_as<Pred, is_equal_t<eve::options<>>> && match(c, category::int8, category::uint8, category::int16, category::uint16))
```

Comment:
I suggest to compare against `decltype(eve::is_equal)` - that's more stable to changes